assets/js/ui.js

```diff
@@ -0,0 +1,212 @@
+            : "carousel-list__item",
+        elementPaginationClass =
+          index === 0
+            ? "pagination-bullet pagination-bullet--active"
+            : "pagination-bullet";
+      if (item.isHighlighted) {
+        let newItem = `
+        <a href="" class="${elementClass}">
+          <figure class="carousel-list__image">
+            <img src="${item.imageUrl}" alt="Title">
+          </figure>
+          <div class="carousel-list__info">
+            <span>${item.title}</span>
+          </div>
+        </a>
+        `;
+        let paginationItem = `<span class="${elementPaginationClass}"></span>`;
+        carouselList.innerHTML += newItem;
+        carouselPagination.innerHTML += paginationItem;
+      }
+    });
+
+    const carouselSlider = new Carousel(carouselContainer);
+    carouselSlider.init();
+  },
+
+  populateProducts(data) {
+    data.length === 0 ? (UI.productsSection.style.display = "none") : "";
+    UI.productList.innerHTML = "";
+    data.forEach((item) => {
+      let newItem = `
+      <a href="" class="products-list__item">
+        <figure class="products-list__image">
+          <img src="${item.imageUrl}" alt="${item.title}">
+        </figure>
+        <h3>${item.title}</h3>
+        <p>${item.text}</p>
+        <div class="products-list__controls">
+          <span class="products-list__price">${item.price} USD</span>
+          <div>
+            <button type="button" class="btn btn--primary btn--xs-full js-edit-product" value="${item.id}">
+              Edit
+            </button>
+            <button type="button" class="btn btn--secondary btn--xs-full js-delete-product" value="${item.id}">
+              Delete
+            </button>
+          </div>
+        </div>
+      </a>
+      `;
+
+      UI.productList.innerHTML += newItem;
+    });
+  },
+
+  clearFields() {
+    UI.inputTitle.value = "";
+    UI.inputPrice.value = "";
+    UI.inputImageUrl.value = "";
+    UI.inputText.value = "";
+    UI.inputHiddenId.value = "";
+    UI.inputIsHighlighted.checked = "";
+  },
+
+  showErrorMsg(state) {
+    const errorEl = `<p class="form-error">Please fill all fields</p>`,
+      containsElement = document.contains(
+        document.querySelector(".form-error")
+      );
+
+    if (state) {
+      if (!containsElement) {
+        UI.modalForm.insertAdjacentHTML("afterbegin", errorEl);
+      }
+    } else {
+      if (containsElement) {
+        document.querySelector(".form-error").remove();
+      }
+    }
+  },
+
+  setButtonState(state) {
+    UI.btnSubmitProduct.textContent =
+      state === "edit" ? "Edit product" : "Add product";
+
+    state === "edit"
+      ? UI.btnSubmitProduct.classList.add("submit-edit")
+      : UI.btnSubmitProduct.classList.remove("submit-edit");
+  },
+
+  checkInputValidation() {
+    if (
+      UI.inputTitle.value !== "" &&
+      UI.inputPrice.value !== "" &&
+      UI.inputImageUrl.value !== "" &&
+      UI.inputText.value !== ""
```

Comment:
Consider using HTML5 validation via `UI.inputTitle.checkValidity()` -- you'll have to specify the `required` attribute. Read up more in MDN ;)